// toasti-toast/dist/toasti-toast.js
import O, { createContext as pr, useContext as gr, useEffect as mr } from "react";
const hr = pr(void 0), Se = () => {
  const n = gr(hr);
  if (!n)
    throw new Error("useToast must be used within a ToastContextProvider");
  return n;
};
var ee = { exports: {} }, $ = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Oe;
function br() {
  if (Oe)
    return $;
  Oe = 1;
  var n = O, f = Symbol.for("react.element"), c = Symbol.for("react.fragment"), l = Object.prototype.hasOwnProperty, v = n.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, R = { key: !0, ref: !0, __self: !0, __source: !0 };
  function x(y, d, k) {
    var E, C = {}, P = null, L = null;
    k !== void 0 && (P = "" + k), d.key !== void 0 && (P = "" + d.key), d.ref !== void 0 && (L = d.ref);
    for (E in d)
      l.call(d, E) && !R.hasOwnProperty(E) && (C[E] = d[E]);
    if (y && y.defaultProps)
      for (E in d = y.defaultProps, d)
        C[E] === void 0 && (C[E] = d[E]);
    return { $$typeof: f, type: y, key: P, ref: L, props: C, _owner: v.current };
  }
  return $.Fragment = c, $.jsx = x, $.jsxs = x, $;
}
var W = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var je;
function yr() {
  return je || (je = 1, process.env.NODE_ENV !== "production" && function() {
    var n = O, f = Symbol.for("react.element"), c = Symbol.for("react.portal"), l = Symbol.for("react.fragment"), v = Symbol.for("react.strict_mode"), R = Symbol.for("react.profiler"), x = Symbol.for("react.provider"), y = Symbol.for("react.context"), d = Symbol.for("react.forward_ref"), k = Symbol.for("react.suspense"), E = Symbol.for("react.suspense_list"), C = Symbol.for("react.memo"), P = Symbol.for("react.lazy"), L = Symbol.for("react.offscreen"), te = Symbol.iterator, De = "@@iterator";
    function Ae(e) {
      if (e === null || typeof e != "object")
        return null;
      var r = te && e[te] || e[De];
      return typeof r == "function" ? r : null;
    }
    var F = n.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function h(e) {
      {
        for (var r = arguments.length, t = new Array(r > 1 ? r - 1 : 0), a = 1; a < r; a++)
          t[a - 1] = arguments[a];
        Ne("error", e, t);
      }
    }
    function Ne(e, r, t) {
      {
        var a = F.ReactDebugCurrentFrame, s = a.getStackAddendum();
        s !== "" && (r += "%s", t = t.concat([s]));
        var u = t.map(function(i) {
          return String(i);
        });
        u.unshift("Warning: " + r), Function.prototype.apply.call(console[e], console, u);
      }
    }
    var Ie = !1, $e = !1, We = !1, ze = !1, Le = !1, ne;
    ne = Symbol.for("react.module.reference");
    function Ye(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === l || e === R || Le || e === v || e === k || e === E || ze || e === L || Ie || $e || We || typeof e == "object" && e !== null && (e.$$typeof === P || e.$$typeof === C || e.$$typeof === x || e.$$typeof === y || e.$$typeof === d || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      e.$$typeof === ne || e.getModuleId !== void 0));
    }
    function Be(e, r, t) {
      var a = e.displayName;
      if (a)
        return a;
      var s = r.displayName || r.name || "";
      return s !== "" ? t + "(" + s + ")" : t;
    }
    function ae(e) {
      return e.displayName || "Context";
    }
    function T(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && h("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case l:
          return "Fragment";
        case c:
          return "Portal";
        case R:
          return "Profiler";
        case v:
          return "StrictMode";
        case k:
          return "Suspense";
        case E:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case y:
            var r = e;
            return ae(r) + ".Consumer";
          case x:
            var t = e;
            return ae(t._context) + ".Provider";
          case d:
            return Be(e, e.render, "ForwardRef");
          case C:
            var a = e.displayName || null;
            return a !== null ? a : T(e.type) || "Memo";
          case P: {
            var s = e, u = s._payload, i = s._init;
            try {
              return T(i(u));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var S = Object.assign, N = 0, oe, ie, se, le, ce, ue, fe;
    function de() {
    }
    de.__reactDisabledLog = !0;
    function Me() {
      {
        if (N === 0) {
          oe = console.log, ie = console.info, se = console.warn, le = console.error, ce = console.group, ue = console.groupCollapsed, fe = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: de,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        N++;
      }
    }
    function Ve() {
      {
        if (N--, N === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: S({}, e, {
              value: oe
            }),
            info: S({}, e, {
              value: ie
            }),
            warn: S({}, e, {
              value: se
            }),
            error: S({}, e, {
              value: le
            }),
            group: S({}, e, {
              value: ce
            }),
            groupCollapsed: S({}, e, {
              value: ue
            }),
            groupEnd: S({}, e, {
              value: fe
            })
          });
        }
        N < 0 && h("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var q = F.ReactCurrentDispatcher, G;
    function Y(e, r, t) {
      {
        if (G === void 0)
          try {
            throw Error();
          } catch (s) {
            var a = s.stack.trim().match(/\n( *(at )?)/);
            G = a && a[1] || "";
          }
        return `
` + G + e;
      }
    }
    var J = !1, B;
    {
      var Ue = typeof WeakMap == "function" ? WeakMap : Map;
      B = new Ue();
    }
    function ve(e, r) {
      if (!e || J)
        return "";
      {
        var t = B.get(e);
        if (t !== void 0)
          return t;
      }
      var a;
      J = !0;
      var s = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var u;
      u = q.current, q.current = null, Me();
      try {
        if (r) {
          var i = function() {
            throw Error();
          };
          if (Object.defineProperty(i.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(i, []);
            } catch (w) {
              a = w;
            }
            Reflect.construct(e, [], i);
          } else {
            try {
              i.call();
            } catch (w) {
              a = w;
            }
            e.call(i.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (w) {
            a = w;
          }
          e();
        }
      } catch (w) {
        if (w && a && typeof w.stack == "string") {
          for (var o = w.stack.split(`
`), b = a.stack.split(`
`), p = o.length - 1, g = b.length - 1; p >= 1 && g >= 0 && o[p] !== b[g]; )
            g--;
          for (; p >= 1 && g >= 0; p--, g--)
            if (o[p] !== b[g]) {
              if (p !== 1 || g !== 1)
                do
                  if (p--, g--, g < 0 || o[p] !== b[g]) {
                    var _ = `
` + o[p].replace(" at new ", " at ");
                    return e.displayName && _.includes("<anonymous>") && (_ = _.replace("<anonymous>", e.displayName)), typeof e == "function" && B.set(e, _), _;
                  }
                while (p >= 1 && g >= 0);
              break;
            }
        }
      } finally {
        J = !1, q.current = u, Ve(), Error.prepareStackTrace = s;
      }
      var A = e ? e.displayName || e.name : "", we = A ? Y(A) : "";
      return typeof e == "function" && B.set(e, we), we;
    }
    function qe(e, r, t) {
      return ve(e, !1);
    }
    function Ge(e) {
      var r = e.prototype;
      return !!(r && r.isReactComponent);
    }
    function M(e, r, t) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return ve(e, Ge(e));
      if (typeof e == "string")
        return Y(e);
      switch (e) {
        case k:
          return Y("Suspense");
        case E:
          return Y("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case d:
            return qe(e.render);
          case C:
            return M(e.type, r, t);
          case P: {
            var a = e, s = a._payload, u = a._init;
            try {
              return M(u(s), r, t);
            } catch {
            }
          }
        }
      return "";
    }
    var V = Object.prototype.hasOwnProperty, pe = {}, ge = F.ReactDebugCurrentFrame;
    function U(e) {
      if (e) {
        var r = e._owner, t = M(e.type, e._source, r ? r.type : null);
        ge.setExtraStackFrame(t);
      } else
        ge.setExtraStackFrame(null);
    }
    function Je(e, r, t, a, s) {
      {
        var u = Function.call.bind(V);
        for (var i in e)
          if (u(e, i)) {
            var o = void 0;
            try {
              if (typeof e[i] != "function") {
                var b = Error((a || "React class") + ": " + t + " type `" + i + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[i] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw b.name = "Invariant Violation", b;
              }
              o = e[i](r, i, a, t, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (p) {
              o = p;
            }
            o && !(o instanceof Error) && (U(s), h("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", a || "React class", t, i, typeof o), U(null)), o instanceof Error && !(o.message in pe) && (pe[o.message] = !0, U(s), h("Failed %s type: %s", t, o.message), U(null));
          }
      }
    }
    var He = Array.isArray;
    function H(e) {
      return He(e);
    }
    function Ke(e) {
      {
        var r = typeof Symbol == "function" && Symbol.toStringTag, t = r && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return t;
      }
    }
    function Xe(e) {
      try {
        return me(e), !1;
      } catch {
        return !0;
      }
    }
    function me(e) {
      return "" + e;
    }
    function he(e) {
      if (Xe(e))
        return h("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", Ke(e)), me(e);
    }
    var I = F.ReactCurrentOwner, Ze = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, be, ye, K;
    K = {};
    function Qe(e) {
      if (V.call(e, "ref")) {
        var r = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function er(e) {
      if (V.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function rr(e, r) {
      if (typeof e.ref == "string" && I.current && r && I.current.stateNode !== r) {
        var t = T(I.current.type);
        K[t] || (h('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', T(I.current.type), e.ref), K[t] = !0);
      }
    }
    function tr(e, r) {
      {
        var t = function() {
          be || (be = !0, h("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: t,
          configurable: !0
        });
      }
    }
    function nr(e, r) {
      {
        var t = function() {
          ye || (ye = !0, h("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: t,
          configurable: !0
        });
      }
    }
    var ar = function(e, r, t, a, s, u, i) {
      var o = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: f,
        // Built-in properties that belong on the element
        type: e,
        key: r,
        ref: t,
        props: i,
        // Record the component responsible for creating this element.
        _owner: u
      };
      return o._store = {}, Object.defineProperty(o._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(o, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: a
      }), Object.defineProperty(o, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: s
      }), Object.freeze && (Object.freeze(o.props), Object.freeze(o)), o;
    };
    function or(e, r, t, a, s) {
      {
        var u, i = {}, o = null, b = null;
        t !== void 0 && (he(t), o = "" + t), er(r) && (he(r.key), o = "" + r.key), Qe(r) && (b = r.ref, rr(r, s));
        for (u in r)
          V.call(r, u) && !Ze.hasOwnProperty(u) && (i[u] = r[u]);
        if (e && e.defaultProps) {
          var p = e.defaultProps;
          for (u in p)
            i[u] === void 0 && (i[u] = p[u]);
        }
        if (o || b) {
          var g = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          o && tr(i, g), b && nr(i, g);
        }
        return ar(e, o, b, s, a, I.current, i);
      }
    }
    var X = F.ReactCurrentOwner, Ee = F.ReactDebugCurrentFrame;
    function D(e) {
      if (e) {
        var r = e._owner, t = M(e.type, e._source, r ? r.type : null);
        Ee.setExtraStackFrame(t);
      } else
        Ee.setExtraStackFrame(null);
    }
    var Z;
    Z = !1;
    function Q(e) {
      return typeof e == "object" && e !== null && e.$$typeof === f;
    }
    function _e() {
      {
        if (X.current) {
          var e = T(X.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function ir(e) {
      {
        if (e !== void 0) {
          var r = e.fileName.replace(/^.*[\\\/]/, ""), t = e.lineNumber;
          return `

Check your code at ` + r + ":" + t + ".";
        }
        return "";
      }
    }
    var Re = {};
    function sr(e) {
      {
        var r = _e();
        if (!r) {
          var t = typeof e == "string" ? e : e.displayName || e.name;
          t && (r = `

Check the top-level render call using <` + t + ">.");
        }
        return r;
      }
    }
    function xe(e, r) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var t = sr(r);
        if (Re[t])
          return;
        Re[t] = !0;
        var a = "";
        e && e._owner && e._owner !== X.current && (a = " It was passed a child from " + T(e._owner.type) + "."), D(e), h('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', t, a), D(null);
      }
    }
    function Ce(e, r) {
      {
        if (typeof e != "object")
          return;
        if (H(e))
          for (var t = 0; t < e.length; t++) {
            var a = e[t];
            Q(a) && xe(a, r);
          }
        else if (Q(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var s = Ae(e);
          if (typeof s == "function" && s !== e.entries)
            for (var u = s.call(e), i; !(i = u.next()).done; )
              Q(i.value) && xe(i.value, r);
        }
      }
    }
    function lr(e) {
      {
        var r = e.type;
        if (r == null || typeof r == "string")
          return;
        var t;
        if (typeof r == "function")
          t = r.propTypes;
        else if (typeof r == "object" && (r.$$typeof === d || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        r.$$typeof === C))
          t = r.propTypes;
        else
          return;
        if (t) {
          var a = T(r);
          Je(t, e.props, "prop", a, e);
        } else if (r.PropTypes !== void 0 && !Z) {
          Z = !0;
          var s = T(r);
          h("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", s || "Unknown");
        }
        typeof r.getDefaultProps == "function" && !r.getDefaultProps.isReactClassApproved && h("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function cr(e) {
      {
        for (var r = Object.keys(e.props), t = 0; t < r.length; t++) {
          var a = r[t];
          if (a !== "children" && a !== "key") {
            D(e), h("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", a), D(null);
            break;
          }
        }
        e.ref !== null && (D(e), h("Invalid attribute `ref` supplied to `React.Fragment`."), D(null));
      }
    }
    function Te(e, r, t, a, s, u) {
      {
        var i = Ye(e);
        if (!i) {
          var o = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (o += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var b = ir(s);
          b ? o += b : o += _e();
          var p;
          e === null ? p = "null" : H(e) ? p = "array" : e !== void 0 && e.$$typeof === f ? (p = "<" + (T(e.type) || "Unknown") + " />", o = " Did you accidentally export a JSX literal instead of a component?") : p = typeof e, h("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", p, o);
        }
        var g = or(e, r, t, s, u);
        if (g == null)
          return g;
        if (i) {
          var _ = r.children;
          if (_ !== void 0)
            if (a)
              if (H(_)) {
                for (var A = 0; A < _.length; A++)
                  Ce(_[A], e);
                Object.freeze && Object.freeze(_);
              } else
                h("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Ce(_, e);
        }
        return e === l ? cr(g) : lr(g), g;
      }
    }
    function ur(e, r, t) {
      return Te(e, r, t, !0);
    }
    function fr(e, r, t) {
      return Te(e, r, t, !1);
    }
    var dr = fr, vr = ur;
    W.Fragment = l, W.jsx = dr, W.jsxs = vr;
  }()), W;
}
process.env.NODE_ENV === "production" ? ee.exports = br() : ee.exports = yr();
var m = ee.exports, ke = {
  color: void 0,
  size: void 0,
  className: void 0,
  style: void 0,
  attr: void 0
}, re = O.createContext && O.createContext(ke), j = globalThis && globalThis.__assign || function() {
  return j = Object.assign || function(n) {
    for (var f, c = 1, l = arguments.length; c < l; c++) {
      f = arguments[c];
      for (var v in f)
        Object.prototype.hasOwnProperty.call(f, v) && (n[v] = f[v]);
    }
    return n;
  }, j.apply(this, arguments);
}, Er = globalThis && globalThis.__rest || function(n, f) {
  var c = {};
  for (var l in n)
    Object.prototype.hasOwnProperty.call(n, l) && f.indexOf(l) < 0 && (c[l] = n[l]);
  if (n != null && typeof Object.getOwnPropertySymbols == "function")
    for (var v = 0, l = Object.getOwnPropertySymbols(n); v < l.length; v++)
      f.indexOf(l[v]) < 0 && Object.prototype.propertyIsEnumerable.call(n, l[v]) && (c[l[v]] = n[l[v]]);
  return c;
};
function Fe(n) {
  return n && n.map(function(f, c) {
    return O.createElement(f.tag, j({
      key: c
    }, f.attr), Fe(f.child));
  });
}
function z(n) {
  return function(f) {
    return O.createElement(_r, j({
      attr: j({}, n.attr)
    }, f), Fe(n.child));
  };
}
function _r(n) {
  var f = function(c) {
    var l = n.attr, v = n.size, R = n.title, x = Er(n, ["attr", "size", "title"]), y = v || c.size || "1em", d;
    return c.className && (d = c.className), n.className && (d = (d ? d + " " : "") + n.className), O.createElement("svg", j({
      stroke: "currentColor",
      fill: "currentColor",
      strokeWidth: "0"
    }, c.attr, l, x, {
      className: d,
      style: j(j({
        color: n.color || c.color
      }, c.style), n.style),
      height: y,
      width: y,
      xmlns: "http://www.w3.org/2000/svg"
    }), R && O.createElement("title", null, R), n.children);
  };
  return re !== void 0 ? O.createElement(re.Consumer, null, function(c) {
    return f(c);
  }) : f(ke);
}
function Rr(n) {
  return z({ tag: "svg", attr: { viewBox: "0 0 1024 1024" }, child: [{ tag: "path", attr: { d: "M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 0 1-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z" } }] })(n);
}
function Pe(n) {
  return z({ tag: "svg", attr: { viewBox: "0 0 1024 1024" }, child: [{ tag: "path", attr: { d: "M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z" } }] })(n);
}
function xr(n) {
  return z({ tag: "svg", attr: { viewBox: "0 0 1024 1024" }, child: [{ tag: "path", attr: { d: "M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm32 664c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V456c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272zm-32-344a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z" } }] })(n);
}
function Cr(n) {
  return z({ tag: "svg", attr: { viewBox: "0 0 1024 1024" }, child: [{ tag: "path", attr: { d: "M955.7 856l-416-720c-6.2-10.7-16.9-16-27.7-16s-21.6 5.3-27.7 16l-416 720C56 877.4 71.4 904 96 904h832c24.6 0 40-26.6 27.7-48zM480 416c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v184c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V416zm32 352a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z" } }] })(n);
}
function Tr(n) {
  return z({ tag: "svg", attr: { viewBox: "0 0 1024 1024", fill: "currentColor", fillRule: "evenodd" }, child: [{ tag: "path", attr: { d: "M799.855 166.312c.023.007.043.018.084.059l57.69 57.69c.041.041.052.06.059.084a.118.118 0 0 1 0 .069c-.007.023-.018.042-.059.083L569.926 512l287.703 287.703c.041.04.052.06.059.083a.118.118 0 0 1 0 .07c-.007.022-.018.042-.059.083l-57.69 57.69c-.041.041-.06.052-.084.059a.118.118 0 0 1-.069 0c-.023-.007-.042-.018-.083-.059L512 569.926 224.297 857.629c-.04.041-.06.052-.083.059a.118.118 0 0 1-.07 0c-.022-.007-.042-.018-.083-.059l-57.69-57.69c-.041-.041-.052-.06-.059-.084a.118.118 0 0 1 0-.069c.007-.023.018-.042.059-.083L454.073 512 166.371 224.297c-.041-.04-.052-.06-.059-.083a.118.118 0 0 1 0-.07c.007-.022.018-.042.059-.083l57.69-57.69c.041-.041.06-.052.084-.059a.118.118 0 0 1 .069 0c.023.007.042.018.083.059L512 454.073l287.703-287.702c.04-.041.06-.052.083-.059a.118.118 0 0 1 .07 0Z" } }] })(n);
}
const wr = {
  success: {
    icon: /* @__PURE__ */ m.jsx(Rr, {}),
    iconClass: "success-icon",
    progressBarClass: "success"
  },
  warning: {
    icon: /* @__PURE__ */ m.jsx(Cr, {}),
    iconClass: "warning-icon",
    progressBarClass: "warning"
  },
  info: {
    icon: /* @__PURE__ */ m.jsx(xr, {}),
    iconClass: "info-icon",
    progressBarClass: "info"
  },
  error: {
    icon: /* @__PURE__ */ m.jsx(Pe, {}),
    iconClass: "error-icon",
    progressBarClass: "error"
  },
  primary: {
    icon: /* @__PURE__ */ m.jsx(Pe, {}),
    iconClass: "error-icon",
    progressBarClass: "primary"
  }
}, Or = ({ message: n, type: f, id: c, timer: l }) => {
  const { icon: v, iconClass: R, progressBarClass: x } = wr[f], y = Se(), d = () => {
    y == null || y.remove(c);
  };
  return mr(() => {
    l && d && setTimeout(() => {
      d();
    }, l);
  }, [l]), /* @__PURE__ */ m.jsxs("div", { className: "flex items-center  shadow-lg rounded-lg p-4 relative w-80 overflow-hidden ", children: [
    /* @__PURE__ */ m.jsx("span", { className: `${R} text-lg`, children: v }),
    /* @__PURE__ */ m.jsx("p", { className: "text-gray-700 text-base font-medium ml-3", children: n }),
    /* @__PURE__ */ m.jsx("button", { className: "cursor-pointer border-none bg-none ml-auto", onClick: d, children: /* @__PURE__ */ m.jsx(re.Provider, { value: { className: "closeButton-icons" }, children: /* @__PURE__ */ m.jsx(Tr, { size: 18, color: "#aeb0d7" }) }) }),
    /* @__PURE__ */ m.jsx("div", { className: "absolute bottom-0 left-0 w-full h-1 bg-opacity-10 bg-gray-100", children: /* @__PURE__ */ m.jsx("div", { className: `toast-progress-bar ${x}` }) })
  ] });
};
function jr(n) {
  switch (n) {
    case "top-left":
      return "top-5 left-5";
    case "top-right":
      return "top-5 right-5";
    case "bottom-left":
      return "bottom-5 left-5";
    case "bottom-right":
      return "bottom-5 right-5";
    case "center":
      return "top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2";
    case "top-center":
      return "top-5 left-1/2 transform -translate-x-1/2 -translate-y-5";
    case "bottom-center":
      return "bottom-5 left-1/2 transform -translate-x-1/2 -translate-y-5";
    case "left-center":
      return "top-1/2 left-5 transform -translate-x-5 -translate-y-1/2";
  }
}
const Sr = () => {
  var c, l;
  const { state: n } = Se(), f = ((c = n.toasts) == null ? void 0 : c.length) - 1;
  return /* @__PURE__ */ m.jsx(m.Fragment, { children: /* @__PURE__ */ m.jsx(
    "div",
    {
      className: `flex flex-col-reverse gap-y-4 z-50 fixed  ${jr(
        ((l = n == null ? void 0 : n.toasts[f]) == null ? void 0 : l.position) ?? "bottom-5 left-5"
      )}`,
      children: n.toasts.map((v) => /* @__PURE__ */ m.jsx(Or, { ...v }, v.id))
    }
  ) });
};
export {
  Sr as ToastiToast,
  Se as useToast
};
